File: app/src/main/java/com/yonder/exercise/ui/detail/IBookDetailRepository.java
package com.yonder.exercise.ui.detail;

import android.arch.lifecycle.LiveData;

import com.yonder.exercise.db.BookModel;
import com.yonder.exercise.network.model.SingleBook;

import io.reactivex.Completable;

/**
 * Created by YusufMac on 02/06/17.
 */

public interface IBookDetailRepository {

    void loadBook(String bookId);

    void updateBook(BookModel event);

    LiveData<SingleBook> getBook();
}

File: app/src/main/java/com/yonder/exercise/ui/detail/IBookDetailRepository.kt
package com.yonder.exercise.ui.detail

import android.arch.lifecycle.LiveData

import com.yonder.exercise.db.BookModel
import com.yonder.exercise.network.model.SingleBook

import io.reactivex.Completable

/**
 * Created by YusufMac on 02/06/17.
 */

interface IBookDetailRepository {
    val book: LiveData<SingleBook>
    fun loadBook(bookId: String)
    fun updateBook(model: BookModel)
}
